alchemist/alchemist-incarnation-biochemistry/src/main/java/it/unibo/alchemist/model/implementations/nodes/CellNodeImpl.java

```diff
@@ -15,29 +15,51 @@
 import java.util.Map;
 import java.util.Set;
 
+import org.apache.commons.math3.util.FastMath;
+
 import com.google.common.collect.MapMaker;
 
 import it.unibo.alchemist.model.implementations.molecules.Biomolecule;
 import it.unibo.alchemist.model.implementations.molecules.Junction;
+import it.unibo.alchemist.model.implementations.positions.Continuous2DEuclidean;
+import it.unibo.alchemist.model.interfaces.CellWithCircularArea;
 import it.unibo.alchemist.model.interfaces.Environment;
-import it.unibo.alchemist.model.interfaces.ICellNode;
+import it.unibo.alchemist.model.interfaces.CellNode;
 import it.unibo.alchemist.model.interfaces.Molecule;
+import it.unibo.alchemist.model.interfaces.Position;
 
 /**
  *
  */
-public class CellNode extends DoubleNode implements ICellNode {
+public class CellNodeImpl extends DoubleNode implements CellNode, CellWithCircularArea {
 
     private static final long serialVersionUID = 837704874534888283L;
 
-    private final Map<Junction, Map<ICellNode, Integer>> junctions = new MapMaker().concurrencyLevel(2).makeMap();
+    private final Map<Junction, Map<CellNode, Integer>> junctions = new MapMaker()
+            .concurrencyLevel(2).makeMap();
```

Comment:
Modify to:
```java
private final Map<Junction, Map<ICellNode, Integer>> junctions = new LinkedHashMap<>();
```

Unsorted maps may have detrimental effects on reproducibility. Please make also sure that no map around is unsorted, nor written concurrently (or accessed and written concurrently). Concurrency is a double-edged sword, with the wrong side sharper.